environment/providers/local/local.go

```diff
@@ -35,10 +36,15 @@ func (prov *provider) GetManagementEndpoint(
 	if endpoint == "" {
 		return nil, types.ErrNotFound
 	}
-	if !strings.HasPrefix(endpoint, "https://") {
-		endpoint = fmt.Sprintf("https://%s", endpoint)
+	port := os.Getenv(portEnv)
+	if port == "" {
+		port = "9440"
 	}
-	addr, err := url.Parse(endpoint)
+	address := fmt.Sprintf("%s:%s", endpoint, port)
+	if !strings.HasPrefix(address, "https://") {
```

Comment:
should we also add a check if endpoint already has port provided?

Reply:
I thought about it and then decided against it. Too many cases to check for it to be worth it (think scheme, IPV6 endpoints, etc.)